src/Compiler/NodeVisitor/CleanupNodeVisitor.php

```diff
@@ -6,28 +6,89 @@
 
 use Efabrica\PHPStanLatte\Resolver\NameResolver\NameResolver;
 use PhpParser\Node;
+use PhpParser\Node\Expr\Assign;
 use PhpParser\Node\Expr\Closure;
 use PhpParser\Node\Expr\ConstFetch;
 use PhpParser\Node\Expr\FuncCall;
+use PhpParser\Node\Expr\MethodCall;
 use PhpParser\Node\Expr\Variable;
 use PhpParser\Node\Stmt\Else_;
 use PhpParser\Node\Stmt\Expression;
 use PhpParser\Node\Stmt\If_;
 use PhpParser\Node\Stmt\TryCatch;
 use PhpParser\NodeTraverser;
 use PhpParser\NodeVisitorAbstract;
+use PhpParser\PrettyPrinter\Standard;
+use PHPStan\Analyser\NodeScopeResolver;
+use PHPStan\Analyser\Scope;
+use PHPStan\Analyser\ScopeContext;
+use PHPStan\Analyser\ScopeFactory;
+use PHPStan\Type\ObjectType;
 
 final class CleanupNodeVisitor extends NodeVisitorAbstract
 {
+    private const METHOD_CALL_CALLER_TYPE = 'method_call_caller_type';
+
     private NameResolver $nameResolver;
 
-    public function __construct(NameResolver $nameResolver)
-    {
+    private ScopeFactory $scopeFactory;
+
+    private Standard $printerStandard;
+
+    private NodeScopeResolver $nodeScopeResolver;
+
+    public function __construct(
+        NameResolver $nameResolver,
+        ScopeFactory $scopeFactory,
+        Standard $printerStandard,
+        NodeScopeResolver $nodeScopeResolver
+    ) {
         $this->nameResolver = $nameResolver;
+        $this->scopeFactory = $scopeFactory;
+        $this->printerStandard = $printerStandard;
+        $this->nodeScopeResolver = clone $nodeScopeResolver;
+    }
+
+    public function beforeTraverse(array $nodes)
+    {
+        $content = $this->printerStandard->prettyPrintFile($nodes);
+        $filename = sys_get_temp_dir() . '/phpstan-latte-' . md5($content) . '.php';
+        file_put_contents($filename, $content);
+        require_once $filename;
+        $scope = $this->scopeFactory->create(ScopeContext::create($filename));
+
+        $nodeCallback = function (Node $node, Scope $scope): void {
+            if ($node instanceof MethodCall) {
+                $type = $scope->getType($node->var);
+                $node->setAttribute(self::METHOD_CALL_CALLER_TYPE, $type);
+            }
+        };
+
+        $this->nodeScopeResolver->processNodes($nodes, $scope, $nodeCallback);
+
+        return null;
     }
 
     public function leaveNode(Node $node)
     {
+        if ($node instanceof If_) {
+            if ($node->cond instanceof Assign) {
+                if ($node->cond->expr instanceof MethodCall) {
```

Comment:
this if is always true for form fields of type CheckboxList (and some other - I have to check return types)
```php
if ($label = $form['checkboxlist']->getLabel()) {
    echo $label;
}
```